src/generated/xinput.rs

```diff
@@ -116,14 +110,16 @@ where
         0,
     ];
     let length_so_far = length_so_far + request0.len();
-    let length_so_far = length_so_far + name.len();
+    let name_bytes = name.serialize();
```

Comment:
Uhm. `name` is an `&[u8]`. This call needlessly copies that to a `Vec` via
https://github.com/psychon/x11rb/blob/787b0955bbf73ecd542f0929a3d48102d63eb587/src/x11_utils.rs#L390-L402

Reply:
I was considering only `CARD8` for this optimization, but `char`, `BYTE` and `void` are also represented with `u8`. I'll fix it.

Reply:
Hm. Seems like xcbgen mapping "everything" to `uint8_t` does have some advantages.

Tomorrow, I'll try to bring some more of the changes here to the python code generator. After having read the diff for `xproto.rs`, I am almost sure I missed stuff and should try that again.

Thanks for your work & good night.